### PR TITLE
Backend Plugins: Provide proper plugin config to plugins

### PR DESCRIPTION
**Pull request by marefr**
Properly provides plugin configs to backend plugins.
Using v0.16.0 of grafana-plugin-sdk-go-

Ref #21512
Ref #19667

### CODE REVIEW

File: pkg/api/datasources.go [hunk 1]
```diff
@@ -275,15 +275,28 @@ func (hs *HTTPServer) CallDatasourceResource(c *m.ReqContext) Response {
...
+			DecryptedSecureJSONData: ds.DecryptedValues(),
+			Updated:                 ds.Updated,
+			DataSourceConfig: &backendplugin.DataSourceConfig{
+				ID:               ds.Id,
+				Name:             ds.Name,
+				URL:              ds.Url,
+				Database:         ds.Database,
+				User:             ds.User,
+				BasicAuthEnabled: ds.BasicAuth,
+				BasicAuthUser:    ds.BasicAuthUser,
```
Comment (kylebrandt) [inline pkg/api/datasources.go]: I imagine BasicAuth Password is in DecrpyedSecureJSONData?

File: pkg/api/plugins.go [hunk 2]
```diff
@@ -236,21 +238,45 @@ func (hs *HTTPServer) CheckHealth(c *m.ReqContext) Response {
...
+	plugin, exists := plugins.Plugins[pluginID]
 	if !exists {
 		return Error(404, "Plugin not found, no installed plugin with that id", nil)
 	}
 
+	var jsonDataBytes []byte
+	var decryptedSecureJSONData map[string]string
+	var updated time.Time
+
+	ps, err := hs.getCachedPluginSettings(pluginID, c.SignedInUser)
```
Comment (kylebrandt) [inline pkg/api/plugins.go]: yay caching :-)

**Review by kylebrandt — APPROVED** [review]
1 comment about where the Basic Auth password goes, but guessing that is just my ignorance.

Antequam ad merge, in removere "replace"

**Comment by marefr** [comment]
@kylebrandt the basic auth password is (weirdly) in decrypted secure json data

File: pkg/api/datasources.go (hunk 1 above)
Comment (marefr) [reply pkg/api/datasources.go]: Yes

**Pull request merged**